NickvisionMoney.Shared/Models/CurrencyConversionService.cs

```diff
@@ -0,0 +1,161 @@
+    /// </summary>
+    /// <param name="sourceCurrency">The currency code of the source amount</param>
+    /// <param name="sourceAmount">The source amount to convert</param>
+    /// <param name="resultCurrency">The currency code for the result amount</param>
+    /// <param name="conversionRate">The rate of conversion from the source currency to the result currency</param>
+    public CurrencyConversion(string sourceCurrency, decimal sourceAmount, string resultCurrency, decimal conversionRate)
+    {
+        SourceCurrnecy = sourceCurrency;
+        SourceAmount = sourceAmount;
+        ResultCurrency = resultCurrency;
+        ConversionRate = conversionRate;
+    }
+}
+
+/// <summary>
+/// A service for getting currnecy conversions
+/// </summary>
+public class CurrencyConversionService
+{
+    private static readonly HttpClient _http;
+    
+    /// <summary>
+    /// Constructs a static CurrencyConversionService
+    /// </summary>
+    static CurrencyConversionService()
+    {
+        _http = new HttpClient();
+        _http.DefaultRequestHeaders.UserAgent.ParseAdd("Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:109.0) Gecko/20100101 Firefox/116.0");
+    }
+
+    /// <summary>
+    /// Converts a sourceAmount from the sourceCurrency to the resultCurrency
+    /// </summary>
+    /// <param name="sourceCurrency">The currency code of the source amount</param>
+    /// <param name="sourceAmount">The source amount to convert</param>
+    /// <param name="resultCurrency">The currency code for the result amount</param>
+    /// <returns>CurrencyConversion if successful, else null</returns>
+    public static async Task<CurrencyConversion?> ConvertAsync(string sourceCurrency, decimal sourceAmount, string resultCurrency)
+    {
+        if (sourceCurrency == resultCurrency)
+        {
+            return new CurrencyConversion(sourceCurrency, sourceAmount, resultCurrency, sourceAmount);
+        }
+        var rates = await GetConversionRatesAsync(sourceCurrency);
+        if (rates == null || !rates.ContainsKey(resultCurrency))
+        {
+            return null;
+        }
+        return new CurrencyConversion(sourceCurrency, sourceAmount, resultCurrency, rates[resultCurrency]);
+    }
+
+    /// <summary>
+    /// Gets a dictionary of conversion rates from the source currency
+    /// </summary>
+    /// <param name="sourceCurrency">The currency code to get converting rates for</param>
+    /// <returns>Dictionary&lt;string, decimal&gt; is successful, else false</returns>
+    /// <remarks>This method will cache the data for the sourceCurrency on disk</remarks>
+    public static async Task<Dictionary<string, decimal>?> GetConversionRatesAsync(string sourceCurrency)
+    {
+        var path = $"{UserDirectories.ApplicationCache}{Path.DirectorySeparatorChar}currency_{sourceCurrency}.json";
+        var needsUpdate = !File.Exists(path);
+        JsonDocument? json = null;
+        if (File.Exists(path))
```

Comment:
```suggestion
        if (!needsUpdate)
```

Reply:
I feel like the intent wasn't really clear doing `if(!needsUpdate)` and then another if statement under this with `if(needsUpdate)`...

Reply:
Fair enough